src/controllers/objecting.entity.name.controller.ts

```diff
@@ -16,6 +22,58 @@ const validators = [
   check(DIVULGE_INFO_FIELD).not().isEmpty().withMessage(ErrorMessages.SELECT_TO_DIVULGE),
 ];
 
+const getExistingPageData = async (session: Session, res: Response, next: NextFunction) => {
+  try {
+    const objection: Objection = await getObjection(session) as Objection;
+    const existingName: string = objection.created_by.fullName;
+    const existingShareIdneity: string = (objection.created_by.shareIdentity === true) ? "yes" : "no";
+    if (existingName && existingShareIdneity) {
+      return res.render(Templates.OBJECTING_ENTITY_NAME, {
+        fullNameValue: existingName,
+        isYesChecked: existingShareIdneity === "yes",
+        isNoChecked: existingShareIdneity === "no",
+        templateName: Templates.OBJECTING_ENTITY_NAME,
+      });
+    } else {
+      return next(new Error("Existing data not present"));
+    }
+  } catch (e) {
+    logger.error(e.message);
+    return next(e);
+  }
+}
+
+const isChangeAnswerFlagPresentSetAndSetToTrue = (session: Session): boolean => {
+  const changeAnswer: boolean = retrieveFromObjectionSession(session, CHANGE_ANSWER_KEY);
+  if (changeAnswer) {
+    return changeAnswer === true;
+  }
+  return false;
+};
+
+/**
+ * GET checks for change flag and renders page
+ * @param req
+ * @param res
+ * @param next
+ */
+export const get = async (req: Request, res: Response, next: NextFunction) => {
+  const session: Session | undefined = req.session as Session;
+  if (session) {
+    if (isChangeAnswerFlagPresentSetAndSetToTrue(session)) {
+      // TODO OBJ-287 handle this more formally.
+      delete retrieveObjectionSessionFromSession(session)[CHANGE_ANSWER_KEY];
+
+      return await getExistingPageData(session, res, next);
```

Comment:
we'd want to show any data they've entered regardless of whether the change flag is present. The change flag should only be used to change the nest page to go to. This way, data will be shown if they use the  'back' button as well.

Reply:
Could be tricky as we cache this in the session initally (pre check company details submit) so getting the objection will return nothing. I thought the aim of this ticket was to base this on the change flag.

Reply:
Added another function for this.